specifications/bcastByz/bcastByz.tla

```diff
@@ -0,0 +1,983 @@
+      BY <1>3, CardThm
+    <2>2 Cardinality(Proc) - Cardinality(Corr) <= T
+      BY <1>7, <2>1, ProcProp, NTFRel
+    <2>3 QED
+      BY <1>3, <1>4, <1>5, <1>6, <2>2, UMCardThm, ProcProp 
+  <1>9 ByzMsgs \subseteq Proc \X M
+    BY  <1>2 DEFS ByzMsgs, M
+  <1>10 QED
+    BY <1>1, <1>2, <1>3, <1>4, <1>5, <1>6, <1>7, <1>8, <1>9 DEF FCConstraints   
+
+THEOREM FCConstraints_Next ==
+  FCConstraints /\ Next => FCConstraints'          
+  <1> SUFFICES ASSUME FCConstraints,
+                      NEW i \in Corr,
+                      Step(i)
+                 PROVE FCConstraints'
+                 BY DEF Next
+  <1>1 Step(i) <=>            
+            \/ Receive(i) /\ UponV1(i)
+            \/ Receive(i) /\ UponNonFaulty(i) 
+            \/ Receive(i) /\ UponAcceptNotSent(i)
+            \/ Receive(i) /\ UponAccept(i)            
+            \/ Receive(i) /\ UNCHANGED << pc, sent, Corr, Faulty >>
+    BY DEF Step    
+  <1>2 CASE Receive(i) /\ UponV1(i)
+    BY <1>2 DEF Receive, UponV1, FCConstraints, ByzMsgs
+  <1>3 CASE Receive(i) /\ UponNonFaulty(i)
+    BY <1>3 DEF Receive, UponNonFaulty, FCConstraints, ByzMsgs
+  <1>4 CASE Receive(i) /\ UponAcceptNotSent(i)
+    BY <1>4 DEF Receive, UponAcceptNotSent, FCConstraints, ByzMsgs 
+  <1>5 CASE Receive(i) /\ UponAccept(i)
+    BY <1>5 DEF Receive, UponAccept, FCConstraints, ByzMsgs  
+  <1>6 CASE Receive(i) /\ UNCHANGED << pc, sent, Corr, Faulty >>
+    BY <1>6 DEF FCConstraints, ByzMsgs   
+  <1>7 QED
+    BY <1>1, <1>2, <1>3, <1>4, <1>5, <1>6 DEF Step      
+  
+    
+(* FCConstraints is an inductive invariant of SpecNoBcast. 
+ * We prove this theorem to ensure that what we do with TLC is correct.
+ *)      
+THEOREM FCConstraints_Init0 == 
+  Init0 => FCConstraints
+  <1> SUFFICES ASSUME Init0
+           PROVE  FCConstraints
+    OBVIOUS       
+  <1>1 Corr \subseteq Proc
+    BY DEF Init0, Proc, IndInv_Unforg_TLC
+  <1>2 Faulty \subseteq Proc
+    BY DEF Init0, Proc, IndInv_Unforg_TLC          
+  <1>3 IsFiniteSet(Corr)
+    BY <1>1, ProcProp, FiniteSetThm2     
+  <1>4 IsFiniteSet(Faulty)
+    BY <1>2, ProcProp, FiniteSetThm2
+  <1>5 Corr \cup Faulty = Proc 
+    BY DEF Init0, Proc, IndInv_Unforg_TLC  
+  <1>6 Faulty = Proc \ Corr
+    BY DEF Init0, Proc, IndInv_Unforg_TLC  
+  <1>7 Cardinality(Corr) >= N - T
+    BY DEF Init0, Proc, IndInv_Unforg_TLC  
+  <1>8 Cardinality(Faulty) <= T          
+    <2>1 Cardinality(Corr) \in Nat
+      BY <1>3, CardThm
+    <2>2 Cardinality(Proc) - Cardinality(Corr) <= T
+      BY <1>7, <2>1, ProcProp, NTFRel
+    <2>3 QED
+      BY <1>3, <1>4, <1>5, <1>6, <2>2, UMCardThm, ProcProp 
+  <1>9 ByzMsgs \subseteq Proc \X M
+    BY  <1>2 DEFS ByzMsgs, M
+  <1>10 QED
+    BY <1>1, <1>2, <1>3, <1>4, <1>5, <1>6, <1>7, <1>8, <1>9 DEF FCConstraints      
+    
+THEOREM FCConstraints_Next0 ==
+  FCConstraints /\ Next0 => FCConstraints'
+    BY FCConstraints_Next DEF Next0, Next      
+      
+THEOREM FCConstraints_Proof == Spec0 => FCConstraints
```

Comment:
Since FCConstraints does not appear under a temporal operator, the assertion only talks about the initial condition. In fact, step <1>2 is trivially true propositionally. Do you mean "Spec0 => []FCConstraints" and accordingly "... => FCConstraints'" in step <1>2?